### ISSUE TITLE
Notifications

### ISSUE THREAD

**Issue by scottietremendous** [concourse/concourse#6600]
#### TLDR

- Notifications is a new concept that would introduce an interface that would be implemented and interpreted by Concourse to support a notification prototype. This prototype would automate actions like getting slack messages or github commit statuses.

#### Problem

- Currently, Concourse users use resources for connecting to Slack, Teams, and other notification tools. This isn't how resources are meant to be used, and it's clumsy to implement them - you need put steps, etc.

#### Proposed Solution

- Build a specified notification prototype to better support this common use-case. 

#### Outcomes

- Remove another vector on which users can misuse the resource abstraction.
- Massively declutters the web ui and pipeline configs because you don't need put steps all over your pipelines.
- Enterprise/Large Scale users will see huge benefits because it makes it easier for developers to efficiently engineer notifications, and makes the UI easier to use too.

#### Current Status

- WIP
- [RFC](https://github.com/concourse/rfcs/pull/28) needs to be finalized
- [Prototypes](https://github.com/concourse/concourse/issues/5719) will need to be implemented before this work can begin

#### References

- WIP [RFC](https://github.com/concourse/rfcs/pull/28)

**Comment by kallisti5** [comment]
something to add.  A special notification path would also allow "state change only" notifications. 
The way things are designed at the moment, you can only produce notifications for "every event" which can get spammy for continuous builds. (especially when you build 5+ architectures :laughing: 

Ideally, it should be possible to adjust what notification events happen.

**State Change only example:**
* Build passed - Notify
* Build passed
* Build failed - Notify
* Build failed
* Build failed
* Build passed - Notify
* Build passed

**State Change and failures only example:**
* Build passed - Notify
* Build passed
* Build passed
* Build failed - Notify
* Build failed - Notify
* Build failed - Notify
* Build passed - Notify
* Build passed
* Build passed

Example of multi-architecture builds with multiple commits:

![spam](https://user-images.githubusercontent.com/100110/125968783-76a90cc9-d68e-4bf8-8e66-d15c22ac242e.png)

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
